docs/internal/release.md

```diff
@@ -0,0 +1,111 @@
+This document describes the release processes and artifacts in `marker`.
+
+# Artifacts
+
+As a result of the engineering effort invested into `marker` we get the following items. Each of them comprises the public API of this project.
+
+## Source packages in `crates.io`
+
+There are library packages published to crates.io such as `marker_api`, `marker_uitest`. They are intended to be statically linked to the users' lint crates.
+
+There are binary packages published to crates.io such as `cargo_marker`, `marker_rustc_driver`. They are intended to be used only in exceptional cases to let users build binaries from source. For example, if `marker` lacks support for some platform where users want to utilize it, then they may leverage `cargo install` to compile `marker`'s component binaries from source. However, 99% of the time users will want to download a precompiled binary.
+
+## Pre-compiled binaries published in a GitHub release
+
+These are the archived binaries such as `cargo-marker-x86_64-unknown-linux-gnu.tar.gz`. These binaries cover as much platforms as possible meaning different combinations of OS (`windows`, `linux`, `macos`), arch (`x86_64`, `aarch64`), libc implementations (`gnu`, `musl`) and their different versions.
+
+Let's review what artifacts there are for a single platform. Take this list as an example.
+
+- `cargo-marker-aarch64-unknown-linux-gnu.tar.gz`
+- `cargo-marker-aarch64-unknown-linux-gnu.zip`
+- `cargo-marker-aarch64-unknown-linux-gnu.sha256`
+
+The `tar.gz` and `zip` archives contain a single `cargo-marker` binary file in them. It doesn't matter what archive the user downloads. They have the choice to select either of them. It is usually common to have only `tar.gz` format for `unix`-like systems, and `zip` for `windows`, but we provide both for the reason described below.
+
+For the cross-platform code it is much more convenient when the same archive format is available for every platform consistently. This means, for example, `cargo-marker` may depend only on the `zip` crate exclusively and expect that `zip` of the  `marker_rustc_driver` is always available for download on all platforms.
+
+> Here is a case of [`terraform`'s](https://developer.hashicorp.com/terraform/downloads) binaries. They are distributed in `zip` for all platforms.
+
+GitHub is very generous with the releases storage limits. They are [almost unlimited](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases#storage-and-bandwidth-quotas). They difinitely want your project to be open-source 😉.
+
+The `sha256` sum is a small file that users may optionally download together with the archive itself to verify the integrity of the archive. It serves as a signature of the artifact to make sure it was downloaded as expected bit-by-bit with what was published effectively detecting corruptions during the download and making it harder to forge the artifact for the malicious actor.
+
+```bash
```

Comment:
I find this script a bit weird, it seems like an abbreviated version of the installation script. It might be better to just link to that one, even if this is just internal documentation.

Instead, we could potentially write:

> <-- region replace-version unstable -->
> This [`install.sh`](https://raw.githubusercontent.com/Veetaha/marker/v0.2.1/install.sh) script, can be used to automatically download and verify Marker's binaries.
> <-- endregion replace-version unstable -->

If we want to keep it, there should be region markers for the version:
```md
<!-- region replace-version unstable -->
...
<!-- endregion replace-version unstable -->
```

Reply:
Agreed, replaced it with the reference to the script.